lib/ui/survey_question/survey_questions_screen.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:survey_flutter_ic/di/di.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_question_views/survey_question_view.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_questions_screen_builder.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_questions_state.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_questions_view_model.dart';
+import 'package:survey_flutter_ic/ui/survey_question/ui_models/survey_questions_ui_model.dart';
+import 'package:survey_flutter_ic/usecases/get_current_survey_use_case.dart';
+import 'package:survey_flutter_ic/utils/context_ext.dart';
+import 'package:survey_flutter_ic/utils/route_path.dart';
+
+final surveyQuestionsViewModelProvider = StateNotifierProvider.autoDispose<
+    SurveyQuestionsViewModel, SurveyQuestionsState>(
+  (_) => SurveyQuestionsViewModel(
+    getIt.get<GetCurrentSurveyUseCase>(),
+  ),
+);
+
+class SurveyQuestionsScreen extends ConsumerStatefulWidget {
+  final String surveyId;
+  final int questionNumber;
+
+  const SurveyQuestionsScreen({
+    super.key,
+    required this.surveyId,
+    required this.questionNumber,
+  });
+
+  @override
+  SurveyQuestionsScreenState createState() => SurveyQuestionsScreenState();
+}
+
+class SurveyQuestionsScreenState extends ConsumerState<SurveyQuestionsScreen> {
+  Map<String, String> arguments(BuildContext context) =>
+      ModalRoute.of(context)?.settings.arguments as Map<String, String>;
+
+  @override
+  void initState() {
+    super.initState();
+    _setUpData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _setupStateListener();
+    return Consumer(
+      builder: (_, ref, __) {
+        final state = ref.watch(surveyQuestionsViewModelProvider);
+        return state.maybeWhen(
+          orElse: () {
+            return const SizedBox.shrink();
+          },
+          submitting: _buildQuestionView,
+          submitted: _buildQuestionView,
+          success: _buildQuestionView,
+          error: (uiModel, _) => _buildQuestionView(uiModel),
+        );
+      },
+    );
+  }
+
+  Widget _buildQuestionView(SurveyQuestionsUIModel uiModel) =>
+      SurveyQuestionView(
+        uiModel: uiModel.question,
+        child: buildAnswer(uiModel.answer),
+        onNextQuestion: () => _nextQuestion(),
+        onSubmit: () => {},
+      );
+
+  void _setUpData() {
+    Future.delayed(Duration.zero, () {
+      ref
+          .read(surveyQuestionsViewModelProvider.notifier)
+          .setUpData(arguments: arguments(context));
```

Comment:
May I ask why do we need to wait `0` second to set up the data @ducbm051291 🤔 It seems a little bit redundant to me

Reply:
We have to add a tiny delay so that we don't get this error :D 
![CleanShot 2023-05-11 at 11 52 54](https://github.com/nimblehq/flutter-ic-david-bruce/assets/24598204/eb565cf3-6139-48c8-92e0-8b5813c5a916)

Reply:
Got it now @ducbm051291 👍 Thanks for clarification